### PR TITLE
ENH: adds --derep_prefix dereplication parameter

### LINKED ISSUE #24: add support for ``--derep_prefix`` when we support vsearch 2.1.1
[A `vsearch` bug](https://github.com/torognes/vsearch/issues/270) was fixed related to this functionality, so we should be able to add this as an option to `dereplicate_sequences`, except that we're currently installing `vsearch` 2.0.3. Version 2.1.1 contains the fix.

### PR DESCRIPTION
**Pull request by colinbrislawn**
First time plug-in development. Feedback wanted!

When finished, resolves #24 

This does not yet have unit tests, I have not done linting, and I'm not sure if it's stylistically consistent. I appreciate _any_ comments and suggestions.

### CODE REVIEW

File: q2_vsearch/_cluster_sequences.py [hunk 1]
```diff
@@ -114,7 +114,8 @@ def _parse_uc(fh):
                       sample_ids=sample_ids)
 
 
-def dereplicate_sequences(sequences: QIIME1DemuxDirFmt
+def dereplicate_sequences(sequences: QIIME1DemuxDirFmt,
+                          prefix: bool=False
```
Comment (thermokarst) [inline q2_vsearch/_cluster_sequences.py]: Can we rename this param to `derep_prefix`, to match the `vsearch`-given name? We have been trying to do that when exposing params on 3rd-party tools like this.

File: q2_vsearch/plugin_setup.py [hunk 2]
```diff
@@ -231,15 +231,22 @@
                       SampleData[SequencesWithQuality] |
                       SampleData[JoinedSequencesWithQuality])
     },
-    parameters={},
+    parameters={
+        'prefix': qiime2.plugin.Bool,
```
Comment (thermokarst) [inline q2_vsearch/plugin_setup.py]: Please see my request above RE renaming this to `derep_prefix`.

**Review by thermokarst — CHANGES_REQUESTED** [review]
This looks awesome @colinbrislawn! I have one minor naming request, inline, otherwise this looks good to me! Thanks so much!

**Comment by colinbrislawn** [comment]
@thermokarst How's it look? I'm excited to merge my first unit test!

**Review by thermokarst — APPROVED** [review]

**Comment by thermokarst** [comment]
Thanks @colinbrislawn! Can you take a few minutes to [update the changelog](https://forum.qiime2.org/t/preview-qiime-2-2018-8-development-changelog/4724?u=thermokarst)? Let me know if you have any questions. Thanks!

**Comment by colinbrislawn** [comment]
I've updated the changelog! Thank you for your support Matt.

**Comment by gregcaporaso** [comment]
Thanks for the contribution @colinbrislawn!

**Pull request merged**